### DISCUSSION TITLE
Frontend exposure on path different from /

### DISCUSSION THREAD

**Discussion by lucascanna** [mia-platform/community#281] [Q&A] [answered]
### Which section?

Marketplace

### Which is the version?

v11.3.0

### Description

Hi guys!

With Mia-Platform Console, frontends are usually exposed on a basepath (configured in the "Endpoints" section).

I am using one of the public Markeplace plugins for building React applications: in particular [this-one](https://github.com/micro-lc/react-template).

When I deploy the microservice, the index.html file is retrived correctly at `my-host.mia-platform.eu/myservice`; however all the other static files are not retrived correctly because the application look at `my-host.mia-platform.eu/index.css` or `my-host.mia-platform.eu/index.js` rather than `my-host.mia-platform.eu/myservice/index.css` or `my-host.mia-platform.eu/myservice/index.js`.

How can I modify the template to tell the application to prepend the basepath `/my-service`?

Thanks in advance for the support.

**Comment by davidebianchi** [comment]
Hi @Lucascanna! The service uses create-react-app under the hood, but it should already have the configuration to handle this behaviour. If you try this:`https://my-host.mia-platform.eu/myservice/`? Maybe, since it is with a relative url, if you set the path without the trailing slash it does not rewrite the urls as you need

**Comment by amountainram** [comment, marked as answer]
Hi Luca,
it really depends on mainly 2 things:
1. how you bundle the plugin -> in the final bundle take a look to the src of static files in places like `style`, `link` and `script` tags.
2. the `base` tag of your `index.html`

In the case of the repo you linked, according to webpack, the application is bundled using the field `homepage` in the `package.json`, thus any static file in the index will appear as `./index.js`. If you type `/my-service` the `./index.js` will be resolved removing `/my-service` due to the trailing slash as @davidebianchi was suggesting.

to remediate either:
- force your api gateway to add a trailing slash, or
- add a base tag on your index html like `<base href="/my-service/" target="_blank" />`
- modify the `homepage` field by setting it to `/my-service/` and build

**Reply by amountainram** [reply]
the 2 latest remediations though will force to rebuild everytime you'd like to change the endpoint in the future

**Comment by lucascanna** [comment]
Hi @davidebianchi and @amountainram,

Thanks a lot for your replies. The trailing slash solved the problem.
I think I will add a redirect on my api-gateway in order to force the trailing slash.
